BASIC BACKEND SERVER REQUESTS

Sidd1600 opened the pull request:
I HAVE PREPARED FILES CONTAINING BASIC SERVER REQUESTS SUCH AS GET,POST,PUT,PATCH AND DELETE AND HAVE ASSEMBLED THEM IN "index.js".I HAVE PREPARED THE SCHEMA OF FOOD ITEMS IN "model.js".

Sidd1600 committed:
Create Read.txt

Sidd1600 committed:
ADDED FILES CONTAINING BASIC SERVER REQUEST

Sidd1600 committed:
Updated index.js

Aryan3902 reviewed (changes requested):
Go through the changes requested

Aryan3902 commented on BACKEND/Read.txt:
Change the filename to readme.md

Aryan3902 commented on BACKEND/index.js:
```diff
@@ -0,0 +1,132 @@
+const app=express();
+const path=require('path');
+const port=80;
+app.set('view engine', 'ejs');
```
Whats the use of this line?

Aryan3902 commented on BACKEND/index.js:
```diff
@@ -0,0 +1,132 @@
+const mongoose = require('mongoose');
+let canteenmenu=require("./model.js");
```
Use const wherever you can

Aryan3902 commented on BACKEND/index.js:
```diff
@@ -0,0 +1,132 @@
+//     id:String
+//   });
+
+  const Menu = mongoose.model('Menu', canteenmenu);
```
Create model where you have defined the scheme

Aryan3902 commented on BACKEND/index.js:
```diff
@@ -0,0 +1,132 @@
+  extended: true
+}));
+
+main().catch(err => console.log(err));
```
Dont use a separate line for catch block, connect like this:

`.connect(url)
    .then(() => {
      console.log("Connected to Database")
        app.listen(PORT);
    })
    .catch(err => { 
      console.log("Error in connecting to database")
        console.log(err)
    });`

BulzEye replied:
@Aryan3902 Actually, he is using the async await syntax for the main function. Thus, his original method was perfectly valid for dealing with errors in the async main() function.

Aryan3902 commented on BACKEND/index.js:
```diff
@@ -0,0 +1,132 @@
+  const Menu = mongoose.model('Menu', canteenmenu);
+
+  
+  const menuitems = Menu.find({}).then((err, posts) => {
```
Add two more folders routers and controllers, routers folder would have two files user-routes and food-routes, controller would have two files  user-controller and food-controller, Router will handle the routes for the http requests and controller will handle the controller function on each request

Aryan3902 commented on BACKEND/index.js:
```diff
@@ -0,0 +1,132 @@
+//     }
+//   });
+  
+app.listen(port,()=>{
```
app.listen will come under mongoose.connect try block

Aryan3902 commented on BACKEND/model.js:
```diff
@@ -0,0 +1,15 @@
+     description: String,
+     id:String
+   });
+ module.exports={
```
dont export these create model here only and export the model

`module.exports = mongoose.model(Menu", canteenmenu );`

Aryan3902 commented on BACKEND/comments.js:
Why do we have this file?

Sidd1600 committed:
Rename Read.txt to Readme.md

Sidd1600 committed:
Create user-routes.js

Sidd1600 committed:
Create food-routes.js

Sidd1600 committed:
Update index.js

Sidd1600 committed:
Update model.js

Sidd1600 commented:
PLEASE GUIDE ME TO DO THE NECESSARY CHANGES  IN FILE "index.js","user-route.js" and "food-route.js" SINCE I AM NOT ABLE TO HANDLE SERVER REQUESTS.

I ALSO TRIED TO IMPLEMENT THE GIVEN CODE 
".connect(url) .then(() => { console.log("Connected to Database") app.listen(PORT); }) .catch(err => {  console.log("Error in connecting to database") console.log(err) });"
AND REMOVED
"main().catch(err => console.log(err));"
BUT AS I REMOVED IT IT STARTED SHOWING ME THE ERROR SHOWN BELOW
"Operation `menus.find()` buffering timed out after 10000ms".

PLEASE HELP

Sidd1600 committed:
Updated index.js with basic server requests

Sidd1600 committed:
Update food-routes.js

Sidd1600 committed:
Update user-routes.js

Sidd1600 commented:
IN LATEST COMMMITS OF "index.js","user-route.js" and "food-route.js",THE ISSUES OF SERVER REQUESTS HAVE BEEN RESOLVED.

BUT ISSUE REGARDING catch() STILL PERSISTS.

Sidd1600 committed:
Create food-controller

Sidd1600 committed:
Create user-controller.js

Sidd1600 committed:
Update food-routes.js

Sidd1600 committed:
Update user-routes.js

Sidd1600 committed:
Update index.js

Sidd1600 committed:
Update model.js

Sidd1600 commented:
I HAVE MODIFIED THE FILES ACCORDING TO MVC MODEL.

BulzEye reviewed (changes requested):
@Sidd1600 Just a few small changes, and we'll be good to go with the PR. Really sorry for the delay.

BulzEye commented on BACKEND/controller/food-controller:
Add .js extension to the file name

BulzEye commented on BACKEND/comments.js:
As stated by @Aryan3902, if this file is needed, fill it with a comment stating its use; otherwise remove it.

BulzEye commented on BACKEND/index.js:
```diff
@@ -0,0 +1,65 @@
+
+// CONNECTING MONGOOSE AND MONGODB
+async function main() {
+  await mongoose.connect('mongodb://127.0.0.1:27017/canteen').then(() => {
```
Change this to what it was before; @Aryan3902 got a bit confused, it was correct before.

BulzEye reviewed (changes requested):
Just make these changes and we'll be all set.

Sidd1600 committed:
Rename food-controller to food-controller.js

Sidd1600 committed:
Delete comments.js

Sidd1600 committed:
Updated index.js
